commands/drawCommand.js

```diff
@@ -22,14 +25,101 @@ export const RARITIES = [
   "Gold Ultra Rare",
 ];
 
+const storingDraws = async (message) => {
+  const response = await axios.get(
+    "https://db.ygoprodeck.com/api/v7/randomcard.php"
+  );
+  let highestRarityIndex = 0;
+  const card = response.data.card_sets;
+
+  card.forEach((element) => {
+    const cardSetRarity = element.set_rarity;
+    const cardSetRarityIndex = RARITIES.indexOf(cardSetRarity);
+    if (highestRarityIndex < cardSetRarityIndex) {
+      highestRarityIndex = cardSetRarityIndex;
+    }
+  });
+
+  message.reply(
+    `You just drew ${response.data.name}, a ${RARITIES[highestRarityIndex]} card`
+  );
+  const attachment = new MessageAttachment(
+    response.data.card_images[0].image_url
+  );
+
+  message.channel.send(attachment);
+
+  const ref = db.ref("users");
+  const userRef = ref.child(message.author.id);
+  userRef.update({
+    lastDrawTime: Date.now(),
+    name: message.author.username,
+  });
+
+  const inventoryRef = userRef.child("inventory");
+  inventoryRef.update({
+    [response.data.id]: {
+      name: response.data.name,
+      image: response.data.card_images[0].image_url,
+      price: response.data.card_prices[0].tcgplayer_price,
+      type: response.data.type,
+      rarity: RARITIES[highestRarityIndex],
+    },
+  });
+};
+
+const payForDraw = async (message) => {
+  const userRef = db.ref(`users/${message.author.id}`);
+  const snapshot = await userRef.once("value");
+  const userData = snapshot.val();
+  const buyADrawMessage = await message.reply(
+    `you can pay $1.50 to draw a card instead of waiting for the time limit. Would you like to buy a draw?`
+  );
+  await buyADrawMessage.react("💵");
+  await buyADrawMessage.react("🚫");
+
+  const filter = (reaction, user) => {
+    return (
+      ["💵", "🚫"].includes(reaction.emoji.name) &&
+      user.id === message.author.id
+    );
+  };
+
+  const collected = await buyADrawMessage.awaitReactions(filter, {
+    max: 1,
+    time: 60000,
+    errors: [],
+  });
+  const reaction = collected.first();
+
+  if (reaction.emoji.name === "💵") {
+    const updatedCurrency = userData.currency - COST_OF_DRAW;
+    userRef.update({ currency: updatedCurrency });
+    storingDraws(message);
+    await buyADrawMessage.delete({ timeout: 1250 });
+  } else if (reaction.emoji.name === "🚫") {
+    const timeDifferenceInMs = DRAW_WAIT_TIME - timeDifference;
+    const timeDifferenceInSeconds = Math.round(timeDifferenceInMs / 1000);
+    const timeDifferenceInMinutes = Math.round(timeDifferenceInMs / 60000);
```

Comment:
you do this in multiple places, so might be worth having a `getReadableTimeDifference(timeDifferenceInMs)` which returns a string including the "minutes" or "seconds"